app/uk/gov/hmrc/agentsubscriptionfrontend/controllers/SubscriptionController.scala

```diff
@@ -101,16 +101,31 @@ class SubscriptionController @Inject()(
     }
   }
 
+  private def updateSessionBeforeSubscribing(registration: Registration)(implicit hc: HeaderCarrier) = {
+    val agencyName = registration.taxpayerName
+    val agencyEmail = registration.emailAddress
+    val agencyAddress = registration.address
+
+    sessionStoreService.cacheAgentSession(AgentSession(registration = Some(
+      Registration(taxpayerName = agencyName,
+        isSubscribedToAgentServices = false,
+        isSubscribedToETMP = false,
+        agencyAddress,
+        agencyEmail))))
+  }
+
   def submitCheckAnswers: Action[AnyContent] = Action.async { implicit request =>
     withSubscribingAgent { agent =>
       agent.withCleanCredsOrSignIn {
         val sjr = agent.getMandatorySubscriptionRecord
         (sjr.businessDetails.utr, sjr.businessDetails.postcode, sjr.businessDetails.registration, sjr.amlsData) match {
           case (utr, postcode, Some(registration), amlsData) =>
-            subscriptionService.handlePartiallySubscribedAndRedirect(agent, sjr.businessDetails.utr, sjr.businessDetails.postcode)(
-              whenNotPartiallySubscribed = subscriptionService
-              .subscribe(utr, postcode, registration, amlsData)
-              .flatMap(redirectSubscriptionResponse(_, agent)))
+                    for {
```

Comment:
Could you please add a comment about why we need to update the session - (for the subscription complete page, after the record is deleted)